### PR TITLE
Fix divide-by-zero for SSCE kernel when normalize factor is zero.

### PR DESCRIPTION
**Pull request by harshithapv**
**Description**: Sparse softmax cross-entropy kernal was returning NaNs when the input all the input tokens were masked causing the normalize_factor = 0. This caused NaNs in Email only domain adaptation pretraining for TNLR_v2 model.

**Motivation and Context**
- Added a check for the case when normalize_factor = 0.

### CODE REVIEW

File: orttraining/orttraining/training_ops/cuda/loss/softmaxcrossentropy_impl.cu [hunk 1]
```diff
@@ -101,7 +101,12 @@ __global__ void _SparseSoftmaxCrossEntropy(
     CUDA_LONG D) {
   CALCULATE_ELEMENTWISE_INDEX_OR_EXIT(i, N);
   CUDA_KERNEL_ASSERT(label_data[i] >= 0 && label_data[i] < D);
-  output_data[i] = -log_prob_data[i * D + label_data[i]] / (*normalize_factor_data);
+  
+  if(*normalize_factor_data == 0){
```
Comment (SherlockNoMad) [inline orttraining/orttraining/training_ops/cuda/loss/softmaxcrossentropy_impl.cu:105]: formatting looks off. please turn on auto formatting and see other code for reference.
